src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -272,15 +278,23 @@ class Host(RESTController):
     @EndpointDoc("List Host Specifications",
                  parameters={
                      'sources': (str, 'Host Sources'),
+                     'facts': (bool, 'Host Facts')
                  },
                  responses={200: LIST_HOST_SCHEMA})
-    def list(self, sources=None):
-        if sources is None:
-            return get_hosts()
-        _sources = sources.split(',')
-        from_ceph = 'ceph' in _sources
-        from_orchestrator = 'orchestrator' in _sources
-        return get_hosts(from_ceph, from_orchestrator)
+    @RESTController.MethodMap(version='1.1')
```

Comment:
Sorry for my ignorance here: What happens to the existing 1.0 implementation? cc @epuertat

Reply:
We already hv [API-versioning](https://github.com/ceph/ceph/pull/35769) implemented, so any changes in endpoint will require some version change, in this case as endpoint provides extra info now and it's stable so 1.1. If you try API call passing 1.0 version in header it'll thow version mismatch error.

Reply:
So in case if this is merged & backported to next pacific release 1.0 won't work for this endpoint.

Reply:
Which mean, we're breaking the API for no reason at all, right? You didn't need to change any tests in `qa/tasks/mgr/dashboard/test_host.py`, cause you didn't add any backward incompatible changes.

Reply:
I mean how are client outside of our git tree supposed to cope with it? Are they supposed to try the 1.1 API and if that doesn't exist, they should use the 1.0 API? That's creating a ton of unnecessary work for everyone writing API clients

Reply:
We update [openapi spec](https://github.com/ceph/ceph/blob/0da1fe3180d6f6d05ed997b19190998977938e41/src/pybind/mgr/dashboard/openapi.yaml#L3214) with the new versioned header too, so as long as u have updated spec I think it won't be any issue to generate API client libraries,etc.
https://github.com/OpenAPITools/openapi-generator
https://swagger.io/tools/swagger-codegen/

Reply:
f> Which mean, we're breaking the API for no reason at all, right? You didn't need to change any tests in `qa/tasks/mgr/dashboard/test_host.py`, cause you didn't add any backward incompatible changes.

The endpoint is still backward compatible, as even if u don't pass the new parameter exposed (`facts`) it still returns the same payload as 1.0 , but it only needs to be passed the correct version. But ok I think now I got your point that <=1.1 should work too in this case 1.0, sorry maybe I got confused initially. So will need some versioning conditions to be handled here.

Reply:
The idea behind API versioning is to **avoid** breaking existing clients. If this PR is backported to pacific, we're actually breaking existing clients cause clients now need to get the 1.1 version.

Reply:
Yes I agree. I've created a new tracker for this https://tracker.ceph.com/issues/52480 ,so will fix it ASAP.

Reply:
@sebastian-philipp as we're merging this to a feature branch, we are giving these more leeway, but you're right: this will break support from legacy API consumers.

Even additive changes to the API [require minor version increases](https://tracker.ceph.com/issues/40909#Versioning-semantics-and-practices), since the main goal of the versioning is to gracefully detect cross-version issues, rather that letting exceptions fly around.

For the methods `a(x)` (1.0) and `a(x, y=default)` (1.1) we need to cover the following cross-version situations:
* 1.0 client calling `a(x_0)` on the 1.1 method `a(x,y)`: :heavy_check_mark: (it'll work due to the way Python deals with optional args)
* 1.1 client calling `a(x_0, y_0)` on the 1.0 method `a(x)`: :x:

Our versioning commitment is that a client consuming a version X.Y will be able to run it against any server providing version X.Z where Y <= Z (minor version/backward support).

Reply:
> For the methods `a(x)` (1.0) and `a(x, y=default)` (1.1) we need to cover the following cross-version situations:

Instead of python functions, I'm more concerned about clients directly accessing the API.

> Our versioning commitment is that a client consuming a version X.Y will be able to run it against any server providing version X.Z where Y <= Z (minor version/backward support).

Thank you.